src/store/store-provider.js

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { reducer } from './reducer';
+import { initialState } from './initial-state';
+
+export const Store = React.createContext();
+
+export const StoreProvider = ({ children }) => {
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const value = { state, dispatch };
```

Comment:
Also -- have you seen any issues (or know of any potential issues) with this pattern?
Passing both `state` and `dispatch` through one `Context`

Reply:
Hi, thanks for spotting that out, I didn't noticed it when I was doing this. I'm thinking the issue might be: The object is re-created everytime this StoreProvider re-renders, so if it re-renders too often, it slows down the performance a bit and cause memory leaks. I've pushed a fix to use `useMemo()` in order to memoise the object.

If there are other issues, it would be perfect if you guys can point me out.

Reply:
@johncalvinroberts After I looked into this line https://github.com/Wiredcraft/omni-admin-ui/blob/d83bb71e4118aad8416e6813fdc9ab57b2c29711/packages/common/src/components/State.js#L87 I finally know why you mentioned this, "official answer" spotted.